src/StoreApi/Routes/Checkout.php

```diff
@@ -177,6 +177,10 @@ protected function get_route_post_response( \WP_REST_Request $request ) {
 		// Persist customer address data to account.
 		$order_controller->sync_customer_data_with_order( $order_object );
 
+		// Fire woocommerce_blocks_checkout_order_processed, should work the same way as woocommerce_checkout_order_processed
+		// But we're opting for a new action because the original ones attaches POST data.
+		do_action( 'woocommerce_blocks_checkout_order_processed', $order_object->get_id(), [], $order_object );
```

Comment:
If we're making a new hook, perhaps it's better to remove the empty / unused `$posted_data` argument. 

This might mean more refactoring on the Subscriptions end, but it more closely matches our intention and what will be passed to extensions implementing the hook.

If we end up needing to pass extra params/options in future we can add an `$options` as the last arg, with specific/known keys.